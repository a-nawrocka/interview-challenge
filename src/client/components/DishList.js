import React from "react";
import Dish from "./Dish";

const DishList = props => {
	return (
		<ul className="item-picker">
			{props.dishes.map(dish => (
				<Dish key={dish.id} dish={dish} />
			))}	
		</ul>
	);
};

export default DishList;