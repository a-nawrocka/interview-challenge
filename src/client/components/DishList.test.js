import React from 'react'
import renderer from 'react-test-renderer';
import DishList from "./DishList";

describe('Given valid array of dish objects', () => {
    it('displays list of dishes', () => {
		const expectedDishes = [
			{
				id: 1,
				name: 'dish one',
				dietaries: ['v', 've', 'df'],
			},
			{
				id: 2,
				name: 'dish two',
				dietaries: ['rsf', 'n!'],
			}
		];

		const component = renderer.create(
			<DishList dishes={expectedDishes} />
		);

		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
    })
})