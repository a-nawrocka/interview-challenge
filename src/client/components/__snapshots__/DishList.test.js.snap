// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given valid array of dish objects displays list of dishes 1`] = `
<ul
  className="item-picker"
>
  <li
    className="item"
  >
    <h2>
      dish one
    </h2>
    <p>
      <span
        className="dietary"
      >
        v
      </span>
      <span
        className="dietary"
      >
        ve
      </span>
      <span
        className="dietary"
      >
        df
      </span>
    </p>
  </li>
  <li
    className="item"
  >
    <h2>
      dish two
    </h2>
    <p>
      <span
        className="dietary"
      >
        rsf
      </span>
      <span
        className="dietary"
      >
        n!
      </span>
    </p>
  </li>
</ul>
`;
